hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/FsDatasetImpl.java

```diff
@@ -452,6 +449,9 @@ private synchronized void activateVolume(
       throw new IOException(errorMsg);
     }
     volumeMap.mergeAll(replicaMap);
+    for (String bp : volumeMap.getBlockPoolList()) {
```

Comment:
Could `storageMap` meet some concurrent-modify issues when move hold lock logic here? Thanks.

Reply:
@Hexiaoqiao Sir, thanks for your review.

- Moving this logic here is to prevent lock leaks when `volumeMap.mergeAll(replicaMap)` failed.
- `storageMap` is a `ConcurrentHashMap` and the modification always need to acquire `synchronized` lock.
- And this VolumeLock can only be used after `volumes.addVolume(ref);`

@MingXiangLi Master, can you help me double-check it? Thanks so much.

Reply:
LGTM here